exec/java-exec/src/main/java/org/apache/drill/exec/proto/helper/QueryIdHelper.java

```diff
@@ -33,7 +33,7 @@ public static String getQueryId(final QueryId queryId) {
 
   public static QueryId getQueryIdFromString(final String queryId) {
     final UUID uuid = UUID.fromString(queryId);
-    return QueryId.newBuilder().setPart1(uuid.getMostSignificantBits()).setPart2(uuid.getLeastSignificantBits()).build();
+    return QueryId.newBuilder().setPart1(uuid.getMostSignificantBits()).setPart2(uuid.getLeastSignificantBits()).setText(queryId).build();
```

Comment:
@vrozov  You're right about the Query ID being identifiable by part1 and part2. However, The PR for DRILL-5305 introduced a change to the protobuf that embeds the Query ID as a text string too. This was so that the profile JSONs could carry that text and a user can rename the file accordingly when attempting to render the profile in the WebUI.
That change causes a situation where the instance of a QueryID created by the foreman of a submitted query carries the text string. But when some other thread might need to instantiate a query ID object based on the query ID's string representation, this value is not set. So, the comparison between a foreman's QueryID instance and a query ID generated by a WebServer thread will not match. This fixes that. In this case, this lookup for a query ID fails despite the entry being there.
https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/work/WorkManager.java#L326

Reply:
@kkhatua I don't see why was it necessary to introduce text id field into the query ID. How for query ID object text field may become different from part1 and part2? This field at best belongs to query profile, not query id.

Reply:
It was introduced as a convenient way of providing it within the query profile's JSON itself, with the intent of having the query ID object text clubbed with part1 & part2, rather than scattered elsewhere within the profile.

Reply:
I think other advantage of having text version of queryId within that class is that it will be computed only once during creation and later just used everywhere as compared to creating new string every time.

Reply:
@vrozov I thought similar is the case with QueryProfile which is defined as protobuf too. But looks like it's only used locally by Foreman or while getting QueryProfile from WebUI. 

I agree with overhead of passing QueryId string over wire with every message as QueryId is used in almost all the rpc messages.

Reply:
Very good point. i didn't anticipate the overhead in the wire transfer protocol for all the rpc messages. I'll make a change to the query profile instead of the query Id.